### ISSUE TITLE
next update plan, feel free to give opinion

### ISSUE THREAD

**Issue by Linaqruf** [Linaqruf/kohya-trainer#138]
0. ~~Synch with main repo~~
1. ~~Delete sample prompt config, because users often forget to run the cell. In the next update, you can edit the sample prompt template by directly editing `sample_prompt.txt`.~~
2. Explain the dreambooth method:
- ~~You don’t need to add instance token and class token if you train dreambooth with captions (all token fields will be empty by default).~~
- ~~You need to add instance token and class token if you set `caption_extension` to none.~~
3. Prune args that are not commonly used, e.g. `caption dropout`. You can still add them by editing `config_file.toml` anyway.
4. ~~Delete the cell for compressing file before deployment, because it’s rarely used.~~
5. ~~Explain how to train with multiple concepts using the latest `dataset_config.toml`.~~
6. ~~Update portable webui to match `cagliostro colab ui`.~~
7. ~~Fix visualize cell.~~
8. Maybe do some minor edits for `fast kohya trainer`.

**Comment by anothertal3** [comment]
First, let me thank you for your efforts! It's only thanks to you and your colab scripts that I've been able to experiment with LoRAs at all. Your descriptions and default values are very helpful for a beginner like me.

Regarding your suggestions, I like the ideas, especially (2) and (5). 

Some of my own input:

1. Please continue to keep those detailed annotations for new functionalities as well. I find it difficult to enter the correct settings for a basic starting point since v14. Especially when it comes to newer functionalities such as loha, locon or optimizers such as DAdaption. For example: I'm still wondering whether `Additional arguments for optimizer, e.g: ["decouple=true","weight_decay=0.6"]` is hinting at DAdaption as I've read about completely other values (0.001) on other sites.

2. I tried to use the Tensorboard to find useful epochs for testing after a run. Apparently, it's not that easy with loss/average or loss/epoch only. I found the following document https://github.com/victorchall/EveryDream2trainer/blob/main/doc/VALIDATION.md and if I understand it correctly then there's a setting `stabilize_training_loss` which can be activated to get another useful log type. However, I am uncertain whether this is even applicable to your training scripts? At least I wasn't able to find the correct place to set it up.

**Comment by Linaqruf** [comment]
> First, let me thank you for your efforts! It's only thanks to you and your colab scripts that I've been able to experiment with LoRAs at all. Your descriptions and default values are very helpful for a beginner like me.
> 
> Regarding your suggestions, I like the ideas, especially (2) and (5).
> 
> Some of my own input:
> 
> 1. Please continue to keep those detailed annotations for new functionalities as well. I find it difficult to enter the correct settings for a basic starting point since v14. Especially when it comes to newer functionalities such as loha, locon or optimizers such as DAdaption. For example: I'm still wondering whether `Additional arguments for optimizer, e.g: ["decouple=true","weight_decay=0.6"]` is hinting at DAdaption as I've read about completely other values (0.001) on other sites.
> 2. I tried to use the Tensorboard to find useful epochs for testing after a run. Apparently, it's not that easy with loss/average or loss/epoch only. I found the following document https://github.com/victorchall/EveryDream2trainer/blob/main/doc/VALIDATION.md and if I understand it correctly then there's a setting `stabilize_training_loss` which can be activated to get another useful log type. However, I am uncertain whether this is even applicable to your training scripts? At least I wasn't able to find the correct place to set it up.

Hi, yeah `["decouple=true","weight_decay=0.6"]` are used for `DAdaptation`, but `weight_decay` is usable for other `optimizer_type` AFAIK, I usually use `["weight_decay=0.6"]` with `AdamW8bit` optimizer and the result are pretty good. Thank you for telling me detailed description is a good thing, I always try to simplify everything because people find it hard to run my notebook even though I love descriptive instruction more. 😄

For validation loss implementation, this is more Kohya's field than mine. Probably you can request the feature in https://github.com/kohya-ss/sd-scripts issue tab. But I don't think it's good to use validation loss for training Stable Diffusion model.

**Comment by Linaqruf** [comment]
Already updated to latest version.

> I'm super excited for the new update! For me, I would like to see more detailed explanations of things like lr_warmup_steps, lr_scheduler_power, and the differences between lora networks and optimizers etc.

The notebook is already updated today 😄, I added some small note for important things like LoRA networks, but for other things I think it's community task to find out. Because I'm afraid I just talking about my preset instead of explaining anything.

**Comment by anothertal3** [comment]
Great! Thanks for the update and thanks for taking my input into account. I will test it as soon as possible.

> Thank you for telling me detailed description is a good thing, I always try to simplify everything because people find it hard to run my notebook even though I love descriptive instruction more. 😄

You're very welcome. Stable Diffusion is progressing insanely fast at the moment and it's hard to keep up. It's even harder to find information or configuration suggestions for the most recent developments. So, any additional input and recommendation is much appreciated :)

By the way: I quite like your idea of having short summaries in the notebook while explaining the details in your new (?) mini guide.

> Hi, yeah ["decouple=true","weight_decay=0.6"] are used for DAdaptation, but weight_decay is usable for other optimizer_type AFAIK, I usually use ["weight_decay=0.6"] with AdamW8bit optimizer and the result are pretty good. 

Hm, I had some very nice results with DAdaption (during V13, I think). But I didn't really configure anything besides the learning rates (1.0 and 0.5) so it was rather a lucky accident. Unfortunately, I haven't been able to produce a similar success since.
I assumed that it might have something to do with requiring different optimizer arguments. I found [the following information](https://rentry.org/59xed3#optimizers) suggesting a `weight_decay` of 0.01, which didn't seem to improve anything.

But all of that is way beyond my knowledge, I'm afraid.

Anyway, I didn't know that `weight_decay` could prove useful even with AdamW8bit. I will make an attempt using your suggested setting of 0.6. 

(Btw: not sure if it's important but I always experiment with realistic Loras, never with NAI-based ones).

> For validation loss implementation, this is more Kohya's field than mine. Probably you can request the feature in https://github.com/kohya-ss/sd-scripts issue tab. But I don't think it's good to use validation loss for training Stable Diffusion model.

I'm not sure if I understand the complexities correctly. It appears to me that there could be a way of training which automatically stops when a certain loss is achieved. But that's seems to be quite difficult.

Therefore, I was actually more interested in this information:
![image](https://user-images.githubusercontent.com/8721770/226995196-b0a0143b-9c79-448b-870f-8a6174d6ef98.png)

That's why I assumed that there might be an existing argument which could be set in order to activate this chart which may or may not help to identify the perfect epoch/step before overfitting begins.

**Comment by Linaqruf** [comment]
I'll try to use `toml` approach instead of `.txt` for sample prompt, so it will be pretty easy to edit it instead of txt, probably in the next update.
